tests/conftest.py

```diff
@@ -145,8 +145,68 @@ def global_execution_env():
             objective = client.add_objective(spec)
             objectives.append(objective)
 
-        assets = _TestAssets(datasets=datasets, objectives=objectives)
-        yield f, assets, n
+        assets = _DataEnv(datasets=datasets, objectives=objectives)
+        return assets
+
+
+@pytest.fixture
+def data_env_1(default_data_env, client_1):
+    """Fixture with pre-existing assets in first node."""
+    return default_data_env.filter_by(client_1.node_id)
+
+
+@pytest.fixture
+def data_env_2(default_data_env, client_2):
+    """Fixture with pre-existing assets in second node."""
+    return default_data_env.filter_by(client_2.node_id)
+
+
+@pytest.fixture
+def default_dataset(data_env_1):
+    """Fixture with pre-existing dataset in first node."""
+    return data_env_1.datasets[0]
```

Comment:
Since this is exactly the same as `default_dataset_1`, why not use the `default_dataset_1` fixture and return it instead of extracting it from data_env_1 ? This is pretty much the same thing but my thinking is that it'd make the chain of fixtures more explicit.

Reply:
You're right!